whylog/teacher/__init__.py

```diff
@@ -0,0 +1,70 @@
+class Interval(object):
+    """
+    Represents interval in line.
+    """
+
+    def __init__(self, start_offset, length, abstract_line):
+        self.start_offset = start_offset
+        self.length = length
+        self.abstract_line = abstract_line
+
+
+class Rule(object):
+    """
+    Represents rule entering by user.
+    """
+
+    def __init__(self, effect):
+        self.effect = effect
+        self.causes = set()
+
+
+class Teacher(object):
+    """
+    Enable teaching new rule. One Teacher per one entering rule.
+    """
+
+    def __init__(self, effect, causes=None):
+        self._rule = Rule(effect)
+        if causes is not None:
```

Comment:
I understand now. You mean `causes=None` is to change.  But why it can't be an empty `set` ?